Styled everything, added some knick-knacks

brennan-obc opened the pull request:
Went too long between commits
-discovered on the last day that the app wasn't broken because it was incomplete, it was broken due to a small typo. 'character' was spelled 'charachter' in half the project, which I remedied. I only got a day to style the populated app, but I did numerous tweaks.
- created logo w/ stock images in GIMP, added to Navbar
- created custom radio buttons (instead of having label next to blue dot, the button is the label)
- added hover, selected, not selected, etc. effects to all buttons -integrated search button into search bar
- made search button unclickable (with visual cues) until search field is filled out and realm is selected
- added stylesheets for most components, made everything look "good" (subjectively speaking of course). All main fonts and almost all colors are stored in variables in index.css, so it can all be easily tweaked.
- added randomized dog breed placeholder to search (randomly selects from array of hundreds of dog breeds)
- made more responsive (adjusting React-Bootstrap grid)
- mobile: made scrollbar disappear on scroll down, reappear on scroll up (vanilla TSX & CSS)
- added unused (commented out) "scan effect" background. See comments atop App.tsx for details
- I'm sure there is more, but that's the gist of it!

brennan-obc committed:
Styled everything, added some knick-knacks

Went too long between commits
-discovered on the last day that the app wasn't broken because it was incomplete, it was broken due to a small typo. 'character' was spelled 'charachter' in half the project, which I remedied. I only got a day to style the populated app, but I did numerous tweaks.
- created logo w/ stock images in GIMP, added to Navbar
- created custom radio buttons (instead of having label next to blue dot, the button is the label)
- added hover, selected, not selected, etc. effects to all buttons
-integrated search button into search bar
- made search button unclickable (with visual cues) until search field is filled out and realm is selected
- added stylesheets for most components, made everything look "good" (subjectively speaking of course). All main fonts and almost all colors are stored in variables in index.css, so it can all be easily tweaked.
- added randomized dog breed placeholder to search (randomly selects from array of hundreds of dog breeds)
- made more responsive (adjusting React-Bootstrap grid)
- mobile: made scrollbar disappear on scroll down, reappear on scroll up (vanilla TSX & CSS)
- added unused (commented out) "scan effect" background. See comments atop App.tsx for details
- I'm sure there is more, but that's the gist of it!

briansimoni commented on src/features/search/Search.tsx:
```diff
@@ -126,27 +148,54 @@ function Search() {
+                  className="custom-form-check"
+                  type="radio"
+                  id="icecrown-radio"
+                  value="Icecrown"
```
```suggestion
                  value="Iceclown"
```
realms === servers

On warmane there are technically three but >90% play on either Blackrock or Icecrown. The big difference is that Blackrock is PVP only. Icecrown gives you the ability to do quests and raids and shit. A common insult to people who play on Icecrown is IceCLOWN

briansimoni reviewed (commented):
This is pretty amazing lol

briansimoni commented on yarn.lock:
We're using npm here, not yarn. This can go away.

briansimoni commented on src/index.css:
We should start thinking about a standard approach to managing CSS. The way that we have it laid out now feels pretty haphazard. Just like any other code, CSS becomes harder to maintain the more disorganized and "dirty" it becomes.

briansimoni commented on setup.txt:
```diff
@@ -0,0 +1,26 @@
+
+## Preperation
+install Node.js & NPM (https://nodejs.org/en/download/)
+install Chocolatey (https://chocolatey.org/install) [optional]
```
I would conjecture that most JavaScript devs these days don't use Windows and don't use Chocolatey.

briansimoni commented on setup.txt:
```diff
@@ -0,0 +1,26 @@
+    -SSH (public-key cryptography, inflexible)
+6. Change directories ['cd warmane-pvp-analytics-ui']
+7. Create new branch ['git branch (new_branch_name)']
+8. Open code editor ['code .']
```
It seems like these notes are more specific to yourself. Definitely keep taking notes like this, but if their for your eyes only don't commit them to GitHub please. I try to prescribe only the minimum tooling i.e. Node.js and npm in this case.

The choice of code editors, git clients, operating system, and similar things should be a personal preference of the developer. #freedom #USA

If we were to commit setup instructions, it should be in markdown format so GitHub can render them for you. We can delete most of the current README.md and use that.

Protip: ctrl+shift+p and type in markdown, and you should see an option to preview in VSCode.

![image](https://user-images.githubusercontent.com/11894935/232247062-7467b378-6294-4730-8af5-369546d7e69a.png)

briansimoni commented on src/components/BackgroundAnimation.tsx:
```diff
@@ -0,0 +1,47 @@
+// While playing around with a gradient generator I found that a white section sweeping from position 0% to 100% sandwiched between gray sections fixed at 0% and 100% looked kind of like a sci-fi scan effect. Hard to explain, but if you check it out you'll see what I mean. I was thinking about implementing it after a user hits search, so it looks like it's "scanning" for player data. I realize no ...
```
Let's use some line breaks bro.

brennan-obc replied:
Sorry I've got line breaks turned on in my editor and never even thought of that.

briansimoni commented on src/components/BackgroundAnimation.tsx:
```diff
@@ -0,0 +1,47 @@
+  children: React.ReactNode;
+}
+
+function BackgroundAnimation({ children }: BackgroundAnimationProps) {
```
I tried this thing out and there might be some potential here. I didn't really understand where exactly I could use it such that it would render in a not weird way.

It does potentially seem like it would fit in it's own npm package. For instance, I'm using [react-data-table-component](https://www.npmjs.com/package/react-data-table-component)

briansimoni replied:
I think there is also merit in having _less_ code laying around. I know I'm not setting a good example with dead components like `<CoolThing>` still laying around, but in my opinion, the more code that exists, the longer it takes to understand what is going on, the higher the complexity, and the higher risk for something to break.

briansimoni commented on src/components/BackgroundAnimation.tsx:
```diff
@@ -0,0 +1,47 @@
+  const generateKeyframes = () => {
+    let keyframes = "";
+
+    for (let i = 0; i <= 100; i++) {
+      keyframes += `
+      ${i}% {
+        background: linear-gradient(90deg, rgba(100,100,106,1) 0%, rgba(253,253,255,0.95) ${i}%, rgba(100,100,96,1) 100%);
+      }`;
+    }
+
+    return keyframes;
+  };
+
+  useEffect(() => {
+    const keyframesElement = document.createElement("style");
+    keyframesElement.id = "gradient-keyframes";
+    keyframesElement.innerHTML = `
+    @keyframes gradientAnimation {
+      ${generateKeyframes()}
+    }
+    `;
```
It seems like the CSS that this generates will be the same every time. Why not just add it to `BackgroundAnimation.css`?

brennan-obc replied:
I was trying to avoid manually incrementing the position of the middle color from 0% to 100% but I suppose I could log the function results (or something similar) and include them in the CSS, would definitely result in cleaner TSX

briansimoni commented on src/components/assets/dog-search.png:
If we are only using dog-search-v2 can we delete this one?

briansimoni commented on src/features/search/Search.tsx:
```diff
@@ -126,27 +148,54 @@ function Search() {
+                  htmlFor="blackrock-radio"
+                  className={`radio-label blackrock${
+                    realm === "Icecrown" ? " anti-selected-realm" : ""
+                  }${hoveredRealm === "Icecrown" ? " anti-hovered-realm" : ""}`}
```
Two ternary statements making up an interpolated string all wrapped inside `{}`.  I'm not a big fan of this.

The search component and the levers that it can pull become complicated quickly. I know what this code does, but we might want to consider moving logic like this into it's own component.

And then maybe we could use it like `<SearchLabel realm={realm} htmlFor="blackrock-radio"/>`

Soon enough you are going to need to understand this redux shit because we may need to do a bit of rewiring of the search logic. If we can isolate some of this aesthetic code elsewhere it might make it easier to focus on the more complicated parts of search.

briansimoni commented on src/App.css:
```diff
@@ -1,12 +1,86 @@
+}
+
+h1 {
+  font-family: var(--title-font);
```
My CSS skills have somewhat decayed so I don't even know what a lot of these rules even do.

briansimoni reviewed (approved):
Things look good. I think we can merge as is, but I'll give you the day to fix things or address the comments if you would like.

You added a ton of CSS. I don't really understand a lot of it so it's hard to determine if it's good or bad. A discord call where you explain it might be the best to bridge the gap here.

If you haven't yet, you may want to Google what makes a good PR or ask ChatGPT. See https://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/ for an example.

I'm calibrating my level of criticism here to balance your learning speed, project delivery time, and code quality. I'm putting much less weight on code quality here than I would at work.

brennan-obc committed:
Comments addressed

Search Aesthetics:
- created SearchLabel.tsx, which contains all the styling related TypeScript from Search.tsx
Refactored CSS:
- all of the component level stylesheets have been deleted, their styles merged into App.css.
- index.css & App.css have been refined to remove DRY violations
Other:
-BackgroundAnimation.tsx/css removed, saved locally
- dog-search.png (v1) deleted
- setup.txt deleted
